src/Dev/CSVParser.php

```diff
@@ -295,7 +294,7 @@ public function __destruct()
     /**
      * @ignore
      */
-    public function rewind()
+    public function rewind(): void
```

Comment:
I'm inclined to just put the return types on now and let things break somewhat unpredictably in this instance.

`#[\ReturnTypeWillChange]` seems like it's just kicking the can down the road, and seems like it assumes a new major version is the time that we will put on the return types.  A new major version of framework is highly unlikely anytime soon, so what would prompt us to actually put the return type in?

I do suppose another way to look at this is use `#[\ReturnTypeWillChange]` for BC compatibility with no intention of ever adding the return types

So I guess it comes down to, do we want to be 'proper' now, or just let this one slide?

Reply:
It seems like the [`#[\ReturnTypeWillChange]`](https://php.watch/versions/8.1/internal-method-return-types#ReturnTypeWillChange) is specifically intended for project in our situation that have to support multiple PHP majors. And since `mixed` is not supported in PHP7.4, we have to at least use `#[\ReturnTypeWillChange]` in the short term for `current()`or `key()` ... might as well use it for all the other ones.

I guess the questions here is how far we want to stretch the definition of semver. I'm thinking the paradigm of avoiding new majors of Silverstripe CMS might be reaching an impasse ... it's the second release in a row where we have to choose between strict adherence to semver and support for a new PHP release. And we'll probably have to do it again for Symfony mailer.

Reply:
> might as well use it for all the other ones
We'll that's certainly the easier of the 2 options.  I'm fine with it.